applications/FluidDynamicsApplication/custom_elements/surface_smoothing_element.h

```diff
@@ -0,0 +1,375 @@
+
+class SurfaceSmoothingElement : public Element
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef Element BaseType;
+
+    ///@}
+    ///@name Pointer Definitions
+    /// Pointer definition of CellNodeProjectionElement
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(SurfaceSmoothingElement);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * Constructor.
+     */
+    SurfaceSmoothingElement(IndexType NewId = 0);
+
+    /**
+     * Constructor using an array of nodes
+     */
+    SurfaceSmoothingElement(IndexType NewId, const NodesArrayType& ThisNodes);
+
+    /**
+     * Constructor using Geometry
+     */
+    SurfaceSmoothingElement(IndexType NewId, GeometryType::Pointer pGeometry);
+
+    /**
+     * Constructor using Properties
+     */
+    SurfaceSmoothingElement(IndexType NewId, GeometryType::Pointer pGeometry, PropertiesType::Pointer pProperties);
+
+    /**
+     * Copy Constructor
+     */
+    SurfaceSmoothingElement(SurfaceSmoothingElement const& rOther);
+
+    /**
+     * Destructor
+     */
+    ~SurfaceSmoothingElement() override;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator.
+    SurfaceSmoothingElement & operator=(SurfaceSmoothingElement const& rOther);
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * ELEMENTS inherited from this class have to implement next
+     * Create and Clone methods: MANDATORY
+     */
+
+    /**
+     * creates a new element pointer
+     * @param NewId: the ID of the new element
+     * @param ThisNodes: the nodes of the new element
+     * @param pProperties: the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+    Element::Pointer Create(IndexType NewId, NodesArrayType const& ThisNodes, PropertiesType::Pointer pProperties) const override;
+
+    /**
+     * creates a new element pointer
+     * @param NewId: the ID of the new element
+     * @param pGeom: the geometry to be employed
+     * @param pProperties: the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+    Element::Pointer Create(IndexType NewId, GeometryType::Pointer pGeom, PropertiesType::Pointer pProperties) const override;
+
+    /**
+     * creates a new element pointer and clones the previous element data
+     * @param NewId: the ID of the new element
+     * @param ThisNodes: the nodes of the new element
+     * @param pProperties: the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+    Element::Pointer Clone(IndexType NewId, NodesArrayType const& ThisNodes) const override;
+
+    /**
+     * this determines the elemental equation ID vector for all elemental
+     * DOFs
+     * @param rResult: the elemental equation ID vector
+     * @param rCurrentProcessInfo: the current process info instance
+     */
+    void EquationIdVector(EquationIdVectorType& rResult, ProcessInfo& CurrentProcessInfo) override;
```

Comment:
you are using the "old" interface, please use the new one, otherwise we will have to update it soon.
Unfortunately there is no official transition guide yet but check the [template element](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/templates/classes/element_template.h) and the PRs I made in the StructuralApp => e.g. #6772 and #6835
Also you can check the baseclass in `element.h`, search for "deprecated" then you will find the "old" and "new" interfaces

In a nutshell, `ProcessInfo` is alsways being passed by `const &` and some fcts have been made `const`

Please ask if sth is unclear

Reply:
Done.